pgdrive/scene_creator/vehicle/base_vehicle.py

```diff
@@ -512,13 +512,14 @@ def _add_chassis(self, pg_physics_world: PGPhysicsWorld):
         self.WIDTH = para[Parameter.vehicle_width]
 
         if self.render:
-            model_path = 'models/ferra/scene.gltf'
-            self.chassis_vis = self.loader.loadModel(AssetLoader.file_path(model_path))
-            self.chassis_vis.setZ(para[Parameter.vehicle_vis_z])
-            self.chassis_vis.setY(para[Parameter.vehicle_vis_y])
-            self.chassis_vis.setH(para[Parameter.vehicle_vis_h])
-            self.chassis_vis.set_scale(para[Parameter.vehicle_vis_scale])
-            self.chassis_vis.reparentTo(self.chassis_np)
+            if self.MODEL is None:
+                model_path = 'models/ferra/scene.gltf'
+                self.MODEL = self.loader.loadModel(AssetLoader.file_path(model_path))
+                self.MODEL.setZ(para[Parameter.vehicle_vis_z])
+                self.MODEL.setY(para[Parameter.vehicle_vis_y])
+                self.MODEL.setH(para[Parameter.vehicle_vis_h])
+                self.MODEL.set_scale(para[Parameter.vehicle_vis_scale])
+            self.MODEL.instanceTo(self.chassis_np)
```

Comment:
what about using a global variable? The self.MODEL is only initialized in __init__ function and can you make sure it will also exist in other BaseVechiels instances?

Reply:
Class variables are shared by all instances. It is more reasonable to store the model in the class variable in oop language. And  I also did this to traffic vehicles' models.